src/lib/systemlib/hysteresis/hysteresis.h

```diff
@@ -49,43 +49,31 @@ namespace systemlib
 class Hysteresis
 {
 public:
-	Hysteresis(bool init_state) :
+	explicit Hysteresis(bool init_state) :
 		_state(init_state),
-		_requested_state(init_state),
-		_hysteresis_time_from_true_us(0),
-		_hysteresis_time_from_false_us(0),
-		_last_time_to_change_state(0)
+		_requested_state(init_state)
 	{}
+	Hysteresis() = delete; // no default constructor
 
-	~Hysteresis()
-	{}
-
-	void set_hysteresis_time_from(const bool from_state, const hrt_abstime new_hysteresis_time_us)
-	{
-		if (from_state == true) {
-			_hysteresis_time_from_true_us = new_hysteresis_time_us;
+	~Hysteresis() = default;
 
-		} else {
-			_hysteresis_time_from_false_us = new_hysteresis_time_us;
-		}
-	}
+	bool get_state() const { return _state; }
 
-	bool get_state() const
-	{
-		return _state;
-	}
+	void set_hysteresis_time_from(const bool from_state, const uint32_t new_hysteresis_time_us);
 
 	void set_state_and_update(const bool new_state);
 
 	void update();
 
 private:
 
+	hrt_abstime _last_time_to_change_state{0};
+
+	uint32_t _time_from_true_us{0};
```

Comment:
So now we wrap at 70 minutes, right? In that case that's scary.

Reply:
It just means you can't configure a time from true/false > ~70 minutes. The internal time change is still a uint64_t.

Reply:
Right but how does that break it's functionality?

Reply:
It doesn't. It's also several thousand times larger than any existing hysteresis usage.

Reply:
I'm still scared about this.

Reply:
We could keep the argument the same width and constrain to uint32 max before storing with an error if you exceed the max (~ 70 minutes). I still think it's nowhere near problematic for anything close to our usage.

Reply:
I really don't think that's worth the optimization. If we add a hack for > 70 minutes then it will never get tested because our testing doesn't fly that long, however, once someone flies with something expensive that can do > 70 minutes, they will be unhappy if it suddenly detects ground or something stupid like that.

Reply:
To be clear there's no issue with flying for more than ~70 minutes, the issue is if the hysteresis threshold in either direction is > ~70 minutes.

Anyway, there's about 7 of these in the actual system at the moment, so it's hardly worth debating. I'll leave this as 64 bit ints for now. It's an easy optimization to come back to if and when needed.

Reply:
Oh, I got it!